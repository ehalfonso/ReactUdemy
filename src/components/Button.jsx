import React from "react";

export const Button = ({ handleClick, title }) => {
  return (
    <>
      <button onClick={handleClick}>{title}</button>
    </>
  );
};
