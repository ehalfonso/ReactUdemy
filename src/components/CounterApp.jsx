import PropTypes from "prop-types";
import { Button } from "./Button";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => {
    setCounter(counter + 1);
  };
  const handleSubtract = () => {
    setCounter(counter - 1);
  };
  const handleReset = () => {
    setCounter(value);
  };
  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <Button handleClick={handleAdd} title={"+1"} />
      <Button handleClick={handleSubtract} title={"-1"} />
      <Button handleClick={handleReset} title={"Reset"} />
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
