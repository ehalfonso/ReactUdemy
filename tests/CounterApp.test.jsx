import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/components/CounterApp";

describe("Test in CounterApp", () => {
  const initialValue = 10;
  test("should do to match with snapshot", () => {
    const { container } = render(<CounterApp />);
    expect(container).toMatchSnapshot();
  });
  test("should show the initial value 100", () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy();
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      "100"
    );
  });
  test("should add with the button +1", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText("11")).toBeTruthy();
  });
  test("should subtract with the button -1", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("-1"));
    // screen.debug();
    expect(screen.getByText("9")).toBeTruthy();
  });
  test("should work the reset button", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("Reset"));
    expect(screen.getByText("10")).toBeTruthy();
  });
});
