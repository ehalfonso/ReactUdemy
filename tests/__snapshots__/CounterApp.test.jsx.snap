// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in CounterApp should do to match with snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2 />
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
